#15 FE 이슈 상세 UI 실제 데이터 출력 / comment 추가 form 구현

rlaqudrnr810 opened the pull request:
## 개요
이슈 상세 UI 구현

## 구현 내용 설명
- 기존의 dummy data 출력에서 API를 활용해 얻어온 실제 정보를 화면에 출력
- comment 추가 레이아웃 구현


## Note
- sidebar form 추가 예정

caribouuuu committed:
Merge pull request #54 from boostcamp-2020/develop

2주차 Day5 버그 수정

rlaqudrnr810 committed:
fix: user-profile 수정

url 인자로 받는 형식으로 수정

rlaqudrnr810 committed:
fix: 이슈 상세 부분 이슈 등록 시간 수정

TitleDetail 부분 이슈 등록 시간 relativeTime 반영

rlaqudrnr810 committed:
fix: IssueDetailContainer.js 변수명 및 dummy data 수정

dummy data 사진 url 수정
dummy.issues[0] -> data 변수명 수정

rlaqudrnr810 committed:
fix: issue profile / comment profile 수정

이슈 등록자 프로필, 코멘트 등록자 프로필 사진 user-profile 사용으로 수정

rlaqudrnr810 committed:
feat: comment 입력 폼 구현

comment 입력 폼 구현
코멘트 입력 시 Comment 버튼 활성화

rlaqudrnr810 committed:
fix: ChangeStatusButton css 수정

ChangeStatusButton css 수정

rlaqudrnr810 committed:
fix: IssueDetail edit 버튼 수정

IssueDetail edit 버튼 수정

rlaqudrnr810 committed:
fix: IssueDetail edit button 수정

Issue Detail Edit button 클릭 시 보이는 테두리 제거

rlaqudrnr810 committed:
fix: detail api 수정

오픈 여부 attribute 추가

rlaqudrnr810 committed:
feat: api 경로 추가

detail api 경로 추가

rlaqudrnr810 committed:
chore: cors 설정

backend cors 설정

rlaqudrnr810 committed:
feat: 실제 데이터 출력

dummy data에서 실제 데이터 출력
hook, fetch api 사용

YimJiYoung commented on frontend/src/containers/detail/IssueDetailContainer.js:
```diff
@@ -9,100 +9,54 @@ import List from './components/List';
-      ],
-    },
-  ],
+const getData = () => {
+  const url = apiUri.detail + document.location.href.split("/")[5];
+  const option = {
+    mode: 'cors',
+    credentials: 'include',
+    method: 'GET',
+  }
+  const [data, setData] = useState([]);
+  const [loading, setLoading] = useState(true);
+  async function fetchUrl() {
+    const response = await fetch(url, option);
+    const json = await response.json();
+    setData(json);
+    setLoading(false);
+  }
+  useEffect(() => {
+    fetchUrl();
+  }, []);
+  return [data, loading];
+}
```
오 .. 다들 로딩 고려해서 하셨네요 .. 훌륭합니다 저는 로딩 생각을 못한 ..
서영님이 만들어주신 useFetch 함수 사용하시면 좋을 것 같습니다 ~

YimJiYoung reviewed (approved):
코드가 더 깔끔해졌네요. 수고하셨습니다 !

rlaqudrnr810 committed:
feat: comment 등록 api 구현

backend/routes/comments 디렉토리 생성
index.js controller.js 생성
해당 이슈에 코멘트 작성 api 구현
backend/routes/index.js에 comments 라우터 추가

rlaqudrnr810 committed:
Merge branch 'develop' into feature/create-comment-be

rlaqudrnr810 committed:
fix: issue detail api 수정

issue detail get API comment 생성 시간 추가

rlaqudrnr810 committed:
feat: api 경로 추가

comment api 경로 추가

rlaqudrnr810 committed:
fix: comment 시간 수정

comment 생성 시간 createdAt으로 수정

rlaqudrnr810 committed:
fix: changestatusbutton 수정

클릭 시 console 로 확인 (아직 미구현)
인자 status -> issue 수정

rlaqudrnr810 committed:
Merge branch 'feature/create-comment-be' of https://github.com/boostcamp-2020/IssueTracker-22 into feature/issue-detail-fe
